create-rust-app_cli/src/main.rs

```diff
@@ -6,181 +6,349 @@ mod qsync;
+                let items = vec!["postgres", "sqlite"];
+                let selection = Select::with_theme(&ColorfulTheme::default())
+                    .items(&items)
+                    .default(0)
+                    .interact_on_opt(&Term::stderr())?;
 
-fn create_project() -> anyhow::Result<()> {
-    let create_opts = CreateOpt::from_args();
-
-    logger::message("Select a database to use:");
-    logger::message("Use UP/DOWN arrows to navigate and SPACE or ENTER to confirm.");
-    let items = vec!["postgres", "sqlite"];
-    let selection = Select::with_theme(&ColorfulTheme::default())
-        .items(&items)
-        .default(0)
-        .interact_on_opt(&Term::stderr())?;
-
-    let backend_database: BackendDatabase = match selection {
-        Some(0) => BackendDatabase::Postgres,
-        Some(1) => BackendDatabase::Sqlite,
-        _ => panic!("Fatal: Unknown backend framework specified."),
+                match selection {
+                    Some(0) => BackendDatabase::Postgres,
+                    Some(1) => BackendDatabase::Sqlite,
+                    _ => panic!("Fatal: Unknown backend database specified."),
+                }
+            } else {
+                panic!("Fatal: No backend database specified")
+            }
+        }
     };
 
-    logger::message("Select a rust backend framework to use:");
-    logger::message("Use UP/DOWN arrows to navigate and SPACE or ENTER to confirm.");
-    let items = vec!["actix-web", "poem"];
-    let selection = Select::with_theme(&ColorfulTheme::default())
-        .items(&items)
-        .default(0)
-        .interact_on_opt(&Term::stderr())?;
-
-    let backend_framework: BackendFramework = match selection {
-        Some(0) => BackendFramework::ActixWeb,
-        Some(1) => BackendFramework::Poem,
-        _ => panic!("Fatal: Unknown backend framework specified."),
-    };
+    // get the backend framework
+    let backend_framework: BackendFramework = match framework {
+        Some(framework) => framework,
+        None => {
+            if interactive {
+                logger::message("Select a rust backend framework to use:");
+                logger::message("Use UP/DOWN arrows to navigate and SPACE or ENTER to confirm.");
+                let items = vec!["actix-web", "poem"];
+                let selection = Select::with_theme(&ColorfulTheme::default())
+                    .items(&items)
+                    .default(0)
+                    .interact_on_opt(&Term::stderr())?;
 
-    let project_name = create_opts.target;
+                match selection {
+                    Some(0) => BackendFramework::ActixWeb,
+                    Some(1) => BackendFramework::Poem,
+                    _ => panic!("Fatal: Unknown backend framework specified."),
+                }
+            } else {
+                panic!("Fatal: No backend database specified")
+            }
+        }
+    };
```

Comment:
these changes are basically asking the following:
- Was a framework set with cli arguments?
    - if so use it, otherwise are we in interactive mode?
        - if so, get a framework with the interactive TUI, otherwise panic